CMakeLists.txt

```diff
@@ -2,231 +2,560 @@ cmake_minimum_required(VERSION 3.1)
+	add_definitions(-D_DEBUG=1)
 endif()
 
-# shinonome font
-set(SHINONOME_PATH "${CMAKE_CURRENT_SOURCE_DIR}/resources/shinonome")
-set(SHINONOME_SRC_PATH "${CMAKE_CURRENT_SOURCE_DIR}/src")
-set(SHINONOME_SRCS "${SHINONOME_SRC_PATH}/shinonome_gothic.cpp"
-  "${SHINONOME_SRC_PATH}/shinonome_mincho.cpp"
-  "${SHINONOME_SRC_PATH}/shinonome.h")
-find_package(Ruby)
-if(RUBY_FOUND)
-  set(SHINONOME_GENERATOR "${SHINONOME_PATH}/generate_cxx_font.rb")
-  add_custom_command(
-    OUTPUT ${SHINONOME_SRCS}
-    COMMAND "${RUBY_EXECUTABLE}" ARGS "${SHINONOME_GENERATOR}"
-    DEPENDS "${SHINONOME_GENERATOR}"
-    WORKING_DIRECTORY "${SHINONOME_PATH}")
-endif()
+add_definitions(-DUSE_SDL=1)
 
-# endianess check
+# Endianess check
 if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-  include(TestBigEndian)
-  test_big_endian(WORDS_BIGENDIAN)
-  if(WORDS_BIGENDIAN)
-    add_definitions(-D WORDS_BIGENDIAN=1)
-  endif()
+	include(TestBigEndian)
+	test_big_endian(WORDS_BIGENDIAN)
+	if(WORDS_BIGENDIAN)
+		add_definitions(-DWORDS_BIGENDIAN=1)
+	endif()
 endif()
 
-if(CMAKE_BUILD_TYPE MATCHES "Debug")
-  add_definitions(-D _DEBUG=1)
+# Adds a library to the liblist if COND is true and the lib is found
+# When REQ is true the lib is required
+function(find_lib_int LIBNAME COND REQ)
+	if(${COND})
+		list(APPEND EASYRPG_LIB_LIST ${LIBNAME})
+	
+		if(${REQ})
+			find_package(${LIBNAME} REQUIRED)
+		else()
+			find_package(${LIBNAME})
+		endif()
+		
+		string(TOUPPER ${LIBNAME} LIBNAMEUP)
+		include_directories(${${LIBNAMEUP}_INCLUDE_DIR} ${${LIBNAMEUP}_INCLUDE_DIRS})
+		set(NEW_LIBS ${${LIBNAMEUP}_LIBRARY} ${${LIBNAMEUP}_LIBRARIES} PARENT_SCOPE)
+	endif()
+
+	# Export _FOUND variable to parent scope
+	set(${LIBNAME}_FOUND ${${LIBNAME}_FOUND} PARENT_SCOPE)
+endfunction()
+
+function(find_lib_req LIBNAME COND)
+	find_lib_int(${LIBNAME} ${COND} ON)
+	
+	# Export _FOUND variable to parent scope which is global scope
+	set(${LIBNAME}_FOUND ${${LIBNAME}_FOUND} PARENT_SCOPE)
+	set(EASYRPG_PLAYER_LIBRARIES ${EASYRPG_PLAYER_LIBRARIES} ${NEW_LIBS} PARENT_SCOPE)
+endfunction()
+
+function(find_lib LIBNAME COND)
+	find_lib_int(${LIBNAME} ${COND} OFF)
+	
+	# Export _FOUND variable to parent scope which is global scope
+	set(${LIBNAME}_FOUND ${${LIBNAME}_FOUND} PARENT_SCOPE)
+	set(EASYRPG_PLAYER_LIBRARIES ${EASYRPG_PLAYER_LIBRARIES} ${NEW_LIBS} PARENT_SCOPE)
+endfunction()
+
+# liblcf
+option(PLAYER_BUILD_LIBLCF "Instead of detecting liblcf the liblcf repository is cloned into lib/liblcf and built together with the Player. This is convenient for development" OFF)
+set(PLAYER_BUILD_LIBLCF_GIT "https://github.com/EasyRPG/liblcf.git" CACHE STRING "Git repository of liblcf to clone when building liblcf. Requires PLAYER_BUILD_LIBLCF=ON.")
+
+if(PLAYER_BUILD_LIBLCF)
+	# liblcf is built as part of this cmake file
+	set(LIBLCF_PATH "${CMAKE_CURRENT_SOURCE_DIR}/lib/liblcf")
+	find_package(Git REQUIRED)
+	if(NOT EXISTS ${LIBLCF_PATH})
+	execute_process(COMMAND ${GIT_EXECUTABLE} clone
```

Comment:
Because you can't really work with a unshallow clone when you want to edit code but I figured out in the meanwhile that there is an unshallow command for me usecase :D